### ISSUE TITLE
Subprocess not terminating, causing install to hang

### ISSUE THREAD

**Issue by justsostephen** [unbit/uwsgi#1237]
On line 692 of `uwsgiconfig.py`, a call to the function `spcall3` is made with the argument `"%s -v" % CPP`, in an attempt to determine the include path for compilation. On my OS X 10.11.3 system with Xcode 7.2.1, this results in a subprocess call to `cpp -v`.

On line 211, the `spcall3` function waits for the child process to exit cleanly. Unfortunately, on my system the subprocess does not exit at all. If I hit `Ctrl + C`, the `wait` method exits and an exception is raised, causing the script to continue using the fallback include path defined on line 704. I've tested this with the `CPP` constant set to `clang -xc core/clang_fake.c` as well - the same thing happens.

**Comment by xrmx** [comment]
Does this work for you?
```diff --git a/uwsgiconfig.py b/uwsgiconfig.py
index 617d6b3..23e7358 100644
--- a/uwsgiconfig.py
+++ b/uwsgiconfig.py
@@ -689,7 +689,7 @@ class uConf(object):
 
         try:
             add_it = False
-            cpp_include_list = str(spcall3("%s -v" % CPP)).split("\n")
+            cpp_include_list = str(spcall3("%s -v < /dev/null" % CPP)).split("\n")
             for line in cpp_include_list:
                 if line.startswith('#include <...> search starts here:'):
                     add_it = True
```

**Comment by xrmx** [comment]
Patch above does not make any sense since spcall3 already does that. So it looks like there's something wrong on mac os x but i don't have an os x system at hand so am not sure can figure out.

**Comment by justsostephen** [comment]
As you say, `spcall3` already passes `/dev/null` to `stdin`.

From Terminal, `clang -xc core/clang_fake.c -v` exits cleanly. `cpp -v < /dev/null` exits cleanly, too. It's only `cpp -v` that doesn't exit. In the script, the subprocess doesn't seem to exit at all, regardless of the command, as run progress stalls at the `wait` method.

**Comment by xrmx** [comment]
@justsostephen so my patch actually work for you?

**Comment by justsostephen** [comment]
No, running `cpp -v < /dev/null` from a Terminal window exits cleanly, but in the script whether `/dev/null` is passed as part of the argument to `spcall3` or passed directly to `Popen` makes no difference - progress still stalls at the `wait` method.

**Comment by xrmx** [comment]
@justsostephen does it work with my patch applied and caling spcall2 instead of spcall3?

**Comment by justsostephen** [comment]
@xrmx, I'm afraid not.
